### ISSUE TITLE
Convert threshold.cc to python

### ISSUE THREAD

**Issue by KieranBrannigan** [iWumboUWumbo2/Hardsub2SRT#2]
Based on the code, all the functions are openCV functions. This means the performance difference shouldn't be too big to be an issue. See this reply on SO: https://stackoverflow.com/a/13433330/10326441

This would remove the need for installing g++ and openCV as a cpp dependency, much nicer for end-users (including me! :P).

Let me know what you think.

Edit: of course, opencv would still be a dependency for python, but all python dependencies can be exported to a requirements.txt file using pip, and then quickly installed with pip.

**Comment by iWumboUWumbo2** [comment]
Yeah that's a good idea. I was planning on doing too. I just kept the c++ code cause I was trying out tesseract before and used that code there
